pnnl.goss.gridappsd/applications/python/volt_var_control.py

```diff
@@ -783,16 +785,77 @@ def VVCCap(self, t0):
                 if self.SubLinkPwr['curr_pf'] < 0:  # Negative current value - implies capacitive loading
                     if self.BasicConfig['desired_pf'] > 0:  # Capacitve is desired, see if we can do something
                         if -self.BasicConfig['desired_pf'] > self.SubLinkPwr['curr_pf']:
-
-
-
-
-
-
-
-
-
-
-
-
-
+                            pass
+                        
+def comment_remover(text):
+    out = ""
+    for line in text.split("\n"):
+        line = line.partition('#')[0]
+        line = line.rstrip()
+        if line:
+            out += line + "\n"
+    return out
+
+
+if __name__ == '__main__':
+    import argparse
+    import json
+    import os
+    
+    default_config_file = os.path.join(
+        os.path.dirname(__file__), "defaults.json")
+    
+    parser = argparse.ArgumentParser(
+        version=__version__)
+    # all of these arguments are required, however one is set to default to the
+    # defaults.json file in the current application directory.  
+    #
+    # docks for argparser are locate here
+    # https://docs.python.org/2.7/library/argparse.html
+    #
+    # After the call to parser.parse_args the returned object will have open
+    # file references to each of the different json config files.  I am leaveing
+    # comments in the different json files so we have to strip all of those off
+    # before we pass it to the json parser so we can get dictionary objects finally
+    # pass to the constructor of the VVC.
+    parser.add_argument('-d', '--default-config', default=default_config_file,
+                        type=file, dest='defaults')
+    parser.add_argument('-r', '--regulator-config', required=True,
+                        type=file, dest='regulators')
+    parser.add_argument('-c', '--capacitor-config', required=True,
+                        type=file, dest='capacitors')
+    
+    opts = parser.parse_args()
+    
+    # Remove the comments from the passed files.
+    defaults = comment_remover(opts.defaults.read())
+    capacitors = comment_remover(opts.capacitors.read())
+    regulators = comment_remover(opts.regulators.read())
+    
+    print('defaults')
+    print('-'*80)
+    print(defaults)
+    
+    print('capacitors')
+    print('-'*80)
+    print(capacitors)
+    
+    print('regulators')
+    print('-'*80)
+    print(regulators)
+    
+    print('-'*80)
+    print('-'*80)
+    # Create dictionary objects from the json string so the VVC class can deal with
+    # them
+    defaultdict = json.loads(defaults)
+    capacitordict = json.loads(capacitors)
+    regulatordict = json.loads(regulators)
+    
+    # Create the VVC object here
```

Comment:
Note I think validation should be done on instantiation of the class or through an explicit validate method